src/admin/components/forms/field-types/Tabs/index.tsx

```diff
@@ -18,16 +21,61 @@ const TabsField: React.FC<Props> = (props) => {
     fieldTypes,
     path,
     permissions,
+    indexPath,
     admin: {
       readOnly,
       className,
     },
   } = props;
 
+  const { getPreference, setPreference } = usePreferences();
+  const { preferencesKey } = useDocumentInfo();
+
   const isWithinCollapsible = useCollapsible();
-  const [active, setActive] = useState(0);
+  const [activeTabIndex, setActiveTabIndex] = useState<number>(0);
+  const tabsPrefKey = `tabs-${indexPath}`;
+
+  useEffect(() => {
+    const getInitialPref = async () => {
+      const existingPreferences: DocumentPreferences = await getPreference(preferencesKey);
+      const initialIndex = path ? existingPreferences?.fields?.[path]?.tabIndex : existingPreferences?.fields?.[tabsPrefKey]?.tabIndex;
+      setActiveTabIndex(initialIndex || 0)
+    }
+    getInitialPref();
+  }, [path, indexPath])
+
+  const handleTabChange = useCallback((incomingTabIndex: number) => {
```

Comment:
This callback could be async and then not require the contained `handlePref` function, right?

Reply:
This is all set